### PR TITLE
Show error when user is missing setup

### PR DESCRIPTION
**Pull request by johnbradley**
Display warning for users who do not have email templates setup.
Hides `New Delivery` button when user is not setup.

Also changes `data-delivery-table` to use named parameters rather than positional.

Requires https://github.com/Duke-GCB/D4S2/pull/183

### CODE REVIEW

File: app/components/delivery-table.js [hunk 1]
```diff
@@ -8,9 +8,13 @@ const statusFilterFunction = function(status, selected) {
   return status.capitalize() == selected;
 };
 
-const DeliveryTable = Ember.Component.extend({
-  currentDukeDsUser: null,
+export default Ember.Component.extend({
+  transfers: null, // required property duke-ds-project-transfers models
+  currentDukeDsUser: null, // required property current user's duke-ds-user model
```
[the remark this answers is not in the document]
Comment (johnbradley) [reply app/components/delivery-table.js]: Unfortunately the `currentDukeDsUser` and `currentUser` properties take some time to resolve and fail the asserts in `didReceiveAttrs`.

**Comment by johnbradley** [comment]
Added `/setup-instructions` route that user gets redirected to if they do not have email templates setup when in the `/new` routes.

**Comment by dleehr** [comment]
Confirmed that attempting to visit `/deliveries/new` and `/deliveries/new/select-project` directly cause me to be redirected to `/deliveries/setup-instructions` when I have no EmailTemplateSet.

**Comment by dleehr** [comment]
Confirmed that attempting to visit `/deliveries/new` and `/deliveries/new/select-project` directly cause me to be redirected to set

**Comment by dleehr** [comment]
Maybe safari is caching something wrong, but now I've re-enabled my EmailTemplateSet and I still get redirected to the setup-instructions page.

**Comment by johnbradley** [comment]
I swear I tested that last week but I am seeing the same behavior.

**Comment by dleehr** [comment]
Yeah something's wrong here. In Safari (private mode) and chrome, I'm always directed to `/deliveries/setup-instructions`

I confirmed in console that the `current-user` response is:

```
{"users":{"id":2,"username":"dcl9","first_name":"","last_name":"","email":"dan@localhost","setup_for_delivery":true}}
```

**Comment by dleehr** [comment]
Oh the code checks `!currentDukeDsUser.get('setupForDelivery')`, but `setupForDelivery` is on the User model (not DukeDSUser)

**Pull request merged**